src/account.c

```diff
@@ -81,29 +81,40 @@ static int param_u32(uint32_t *v, const struct pl *params, const char *name)
 static int stunsrv_decode(struct account *acc, const struct sip_addr *aor)
 {
 	struct pl srv, tmp;
+	struct uri uri;
 	int err;
 
 	if (!acc || !aor)
 		return EINVAL;
 
+	memset(&uri, 0, sizeof(uri));
 	if (0 == msg_param_decode(&aor->params, "stunserver", &srv)) {
 
 		info("using stunserver: '%r'\n", &srv);
 
-		err = stunuri_decode(&acc->stun_host, &srv);
+		err = uri_decode(&uri, &srv);
 		if (err) {
-			warning("account: decode '%r' failed: %m\n",
+			warning("account: decode '%r' failed (%m)\n",
 				&srv, err);
+			return err;
 		}
+
+		stunuri_decode_uri(&acc->stun_host, &uri);
 	}
 
 	err = 0;
 
 	if (0 == msg_param_exists(&aor->params, "stunuser", &tmp))
 		err |= param_dstr(&acc->stun_user, &aor->params, "stunuser");
+	else if (pl_isset(&uri.user))
+		err |= re_sdprintf(&acc->stun_user, "%H",
+					uri_user_unescape, &uri.user);
 
 	if (0 == msg_param_exists(&aor->params, "stunpass", &tmp))
 		err |= param_dstr(&acc->stun_pass, &aor->params, "stunpass");
+	else if (pl_isset(&uri.password))
+		err |= re_sdprintf(&acc->stun_pass, "%H",
+					uri_password_unescape, &uri.password);
```

Comment:
the uri "password" field is deprecated

better to use a parameter e.g. "password=XXXX" or the credential field
in the STUN uri spec

Reply:
You are completely correct that the "password" uri field is deprecated. But I don't see how to prevent stunserver from being used with this field. Moreover, in [;stunserver=stun:[user:pass]@host[:port]](https://github.com/baresip/baresip/blob/main/docs/examples/accounts) this field is indicated as an example for the stunserver parameter.
Might be worth generating a warning message if this option is used?
`
	else if (pl_isset(&uri.password)) {
		err |= re_sdprintf(&acc->stun_pass, "%H",
					uri_password_unescape, &uri.password);
		warning("account: use of the format "user:pass" in the stunserver field is deprecated\n",
				&srv, err);			
	}
`